src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/TraceLoggingEventSource.cs

```diff
@@ -21,15 +21,19 @@
 using System.Text;
 
 #if ES_BUILD_STANDALONE
+using System.Runtime.CompilerServices;
 namespace Microsoft.Diagnostics.Tracing
 #else
+using Internal.Runtime.CompilerServices;
 namespace System.Diagnostics.Tracing
 #endif
 {
     public partial class EventSource
     {
 #if FEATURE_MANAGED_ETW
-        private byte[] providerMetadata = null!;
+        private byte[]? m_providerMetadata;
+        private protected virtual ReadOnlySpan<byte> ProviderMetadata => m_providerMetadata;
+        private bool HasProviderMetadata => !Unsafe.IsNullRef(ref MemoryMarshal.GetReference(ProviderMetadata));
```

Comment:
Why not just check the Length?

Reply:
To differentiate as when `m_traits != null` it generates a `providerMetadata` local variable but then leaves `m_providerMetadata` as null.

However, thinking about it that's probably a bug so will fix it 😀

Reply:
Ah; I know why I did this, it was for the manifest which can be legitimately `null`, so to differentiate between being set and empty and not set.

Isn't needed for the metadata and I didn't do the manifest (as that requires quite a lot of introspection and my meagre knowledge of navigating Roslyn source trees isn't quite ready for that)